internal/expr/query.go

```diff
@@ -87,17 +90,38 @@ func (pe *PreparedExpr) Query(args ...any) (ce *QueryExpr, err error) {
 		}
 		var val reflect.Value
 		switch tm := typeMember.(type) {
-		case structField:
+		case *structField:
 			val = v.FieldByIndex(tm.index)
-		case mapKey:
+			qargs = append(qargs, sql.Named("sqlair_"+strconv.Itoa(argCount), val.Interface()))
+			argCount++
+		case *mapKey:
 			val = v.MapIndex(reflect.ValueOf(tm.name))
 			if val.Kind() == reflect.Invalid {
 				return nil, fmt.Errorf(`map %q does not contain key %q`, outerType.Name(), tm.name)
 			}
+			if val.Kind() == reflect.Interface {
```

Comment:
As these branches are mutually exclusive, would it be better to use a `switch` here instead?

Reply:
They're not, the first one mutates `val`.